source/interface/azure_iot_mqtt.h

```diff
@@ -0,0 +1,380 @@
+                                             uint16_t usPacketId );
+
+/**
+ * @brief Publishes a message to the given topic name.
+ *
+ * @param[in] xContext Initialized MQTT context.
+ * @param[in] pxPublishInfo MQTT PUBLISH packet parameters.
+ * @param[in] usPacketId packet ID generated by #AzureIoTMQTT_GetPacketId.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_Publish( AzureIoTMQTTHandle_t xContext,
+                                           const AzureIoTMQTTPublishInfo_t * pxPublishInfo,
+                                           uint16_t usPacketId );
+
+/**
+ * @brief Sends an MQTT PINGREQ to broker.
+ *
+ * @param[in] xContext Initialized and connected MQTT context.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_Ping( AzureIoTMQTTHandle_t xContext );
+
+/**
+ * @brief Sends MQTT UNSUBSCRIBE for the given list of topic filters to
+ * the broker.
+ *
+ * @param[in] xContext Initialized MQTT context.
+ * @param[in] pxSubscriptionList List of MQTT subscription info.
+ * @param[in] xSubscriptionCount The number of elements in pSubscriptionList.
+ * @param[in] usPacketId packet ID generated by #AzureIoTMQTT_GetPacketId.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_Unsubscribe( AzureIoTMQTTHandle_t xContext,
+                                               const AzureIoTMQTTSubscribeInfo_t * pxSubscriptionList,
+                                               size_t xSubscriptionCount,
+                                               uint16_t usPacketId );
+
+/**
+ * @brief Disconnect an MQTT session.
+ *
+ * @param[in] xContext Initialized and connected MQTT context.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_Disconnect( AzureIoTMQTTHandle_t xContext );
+
+
+/**
+ * @brief Loop to receive packets from the transport interface. Handles keep
+ * alive.
+ *
+ * @param[in] xContext Initialized and connected MQTT context.
+ * @param[in] ulTimeoutMs Minimum time in milliseconds that the receive loop will
+ * run, unless an error occurs.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_ProcessLoop( AzureIoTMQTTHandle_t xContext,
+                                               uint32_t ulTimeoutMs );
+
+
+/**
+ * @brief Loop to receive packets from the transport interface. Does not handle
+ * keep alive.
+ *
+ * @param[in] xContext Initialized and connected MQTT context.
+ * @param[in] ulTimeoutMs Minimum time in milliseconds that the receive loop will
+ * run, unless an error occurs.
+ *
+ * @return An #AzureIoTMQTTResult_t with the result of the operation.
+ */
+AzureIoTMQTTResult_t AzureIoTMQTT_ReceiveLoop( AzureIoTMQTTHandle_t xContext,
+                                               uint32_t ulTimeoutMs );
```

Comment:
Nit - the names are `Ms` here for milliseconds but IIRC the public API names these `milliseconds`.  Not critical but we should consider having this be consistent E2E through the MS headers, regardless of the naming of underlying MQTT stack.

Reply:
Ack

---
In reply to: [598966514](https://github.com/Azure/azure-iot-middleware-freertos/pull/44#discussion_r598966514) [](ancestors = 598966514)